File: frontend/packages/shared/types/reimbursement.ts
import type { PageQuery } from './common'
import type { ExpenseVO } from './expense'

export interface ReimbursementCreateDTO {
  title: string
  remark?: string
  expenseIds: number[]
  tripId?: number
}

export interface ReimbursementVO {
  id: number
  userId: number
  reimburseNo: string
  title: string
  totalAmount: number
  invoiceCount: number
  itemCount: number
  remark: string
  pdfUrl: string
  mergedPdfUrl: string
  zipUrl: string
  tripId: number
  reimburseStatus: number
  exportCount: number
  exportedAt: string
  emailSent: number
  emailAddress: string
  emailSentAt: string
  receivedAt: string
  status: number
  createdAt: string
  updatedAt: string
  expenses: ExpenseVO[]
}

export interface ReimbursementQueryDTO extends PageQuery {
  reimburseStatus?: number
}

export type ReimbursementStatus = 'generated' | 'exported' | 'received'

export interface ExportOptions {
  type: 'merged_pdf' | 'zip' | 'report_only' | 'email'
  email?: string
}

File: backend/src/main/java/com/qiankubx/module/reimbursement/entity/Reimbursement.java
package com.qiankubx.module.reimbursement.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("t_reimbursement")
public class Reimbursement {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long userId;

    private String reimburseNo;

    private String title;

    private BigDecimal totalAmount;

    private Integer invoiceCount;

    private Integer itemCount;

    private String remark;

    private String pdfUrl;

    private String mergedPdfUrl;

    private String zipUrl;

    private Long tripId;

    private Integer reimburseStatus;

    private Integer exportCount;

    private LocalDateTime exportedAt;

    private Integer emailSent;

    private String emailAddress;

    private LocalDateTime emailSentAt;

    private LocalDateTime receivedAt;

    private Integer status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
